### DISCUSSION TITLE
REST API: add a way to delete check-ins

### DISCUSSION THREAD

**Discussion by stapelberg** [pretix/pretix#3745] [Ideas]
We have a custom frontend with which we check people into our events (using the redeem method).

Sometimes, a check-in is done incorrectly and needs to be undone.

Currently, this is only possible by opening up pretix, locating the check-in and deleting it via the pretix UI.

AFAICT, the API offers no way to delete a check-in. I suggest adding one. Does that sound reasonable?

**Comment by raphaelm** [comment]
I'm torn. On one hand, even though they are not financial information, we desire check-ins in pretix to be pretty "immutable" to give a good data set on real-world actions. On the other hand, we already offer a delete option for all check-ins of a ticket through the backend, so it would be only logical to add this to the API as well.

However, we have (vague!) plans of adding a concept of "revoking" a check-in, which we will need in combination with some hardware setups ~next year (e.g. scanning the ticket unlocks a door, but the check-in gets reverted if the door never gets opened), which would be an even cleaner thing to have than deletion, but we haven't really made a solid concept yet.

**Reply by stapelberg** [reply]
Revoking a check-in sounds like it would work for my use-case as well. Maybe I’ll use a custom patch until that’s available :)
